### PR TITLE
Connect to API

### PR DESCRIPTION
**Pull request by Khaled-AlKhateeb**
##Tasks Done ✔️ 

- Fetch book list from Bookstore API on Books page load.
- Refactored add book and remove book features to persist changes in the server.
- Add book and remove book work in the same way after the refactor.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    });
+    Object.keys(data).forEach((key, i) => {
+      dataArray[i].id = key;
+    });
+    dispatch(fullfiled(dataArray));
+  });
+
+export const postBookItem = (obj) => async (dispatch) => {
+  fetch(APIURLPATH, {
+    method: 'POST',
```
Comment (BertrandConxy) [inline src/redux/books/books.js]: - [x] Good job so far. It is among the requirements to use `createAsyncThunk` here in redux file 😃 👍 
As a hint, your books reducer file should look like the following 👇
```js
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios'; 


export const getBooks = createAsyncThunk(
  'books/getBooks',
  async () => {
    const response = await axios.get(
      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/',
    );
    return response.data;
  },
); 
```

Please implement it in your file too 😉

**Review by BertrandConxy — CHANGES_REQUESTED** [review]
Hi @Khaled-AlKhateeb, 

Good job so far! 👏 👏 
There are some issues that you still need to work on to go to the next project but you are almost there! 🏃‍♂️ 🏃‍♂️ 


## Highlights 💪 
- No linter errors
- correct flow is used
- good PR


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    },
+  }).then(() => dispatch(addBook(obj)));
+};
+
+export const deleteBookItem = (id) => async (dispatch) => {
+  fetch(`${APIURLPATH}/${id}`, {
+    method: 'DELETE',
+    body: JSON.stringify(id),
+  }).then(() => dispatch(removeBook(id)));
+};
```
Comment (mmsesay) [inline src/redux/books/books.js]: - [ ] In addition to the [previous reviewer's comment](https://github.com/Khaled-AlKhateeb/bookstore/pull/4#discussion_r1020787739), please use `createAsyncThunk` on the `postBookItem` and `deleteBookItem` functions. Just as you implemented it on the `fetchBooks`

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi @Khaled-AlKhateeb, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Highlights
 - Good Readme :heavy_check_mark: 
 - No linter error :heavy_check_mark: 
 - Good PR title :heavy_check_mark: 
 - Good PR summary :heavy_check_mark:  

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @Khaled-AlKhateeb, 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**